app/domain/rotation/master_rotator.rb

```diff
@@ -0,0 +1,144 @@
+module Rotation
+  class MasterRotator
+
+    class Resource
+      attr_reader :id
+
+      def initialize(resource_id)
+        @id = resource_id
+      end
+
+      def account
+        @id.split(':')[0]
+      end
+
+      def kind
+        @id.split(':')[1]
+      end
+
+      def name
+        @id.split(':', 3)[2]
+      end
+
+      def renamed(name)
+        self.class.new("#{account}:#{kind}:#{name}")
+      end
+    end
+
+    class ScheduledRotation
+
+      RotatorNotFound = ::Util::ErrorClass.new(
+        "'{0}' is not an installed rotator"
+      )
+
+      attr_reader :resource, :ttl, :rotator_name
+
+      def initialize(resource_id:, ttl:, rotator_name:,
+                     avail_rotators:, secret_model:)
+        @resource = ::Rotation::MasterRotator::Resource.new(resource_id)
+        @ttl = ttl
+        @rotator_name = rotator_name
+        @avail_rotators = avail_rotators
+        @secret_model = secret_model
+        validate!
+      end
+
+      # Really this is a pipeline, but it's hard to express it as such in ruby:
+      #
+      # current_values | name_keys | new_values | resource_id_keys
+      #
+      def new_values
+        resource_id_keys(rotator.new_values(name_keys(current_values)))
+      end
+
+      private
+
+      def current_values
+        @secret_model.latest_resource_values(required_resources)
+        p 'current_values', current_values
```

Comment:
Stray log message. There's at least one more elsewhere too, I don't want to litter the PR pointing them out though :-) That's been indicated as a bad PR practice.